src/app/_components/Form/FormListbox.tsx

```diff
@@ -0,0 +1,64 @@
+import { Listbox } from '@headlessui/react'
+import theme from 'tailwindcss/defaultTheme'
+
+import { FormField, type FormFieldProps } from '@/components/Form/FormField'
+import { type IconProps } from '@/components/Icon'
+import { ListboxButton } from '@/components/Listbox/ListboxButton'
+import {
+  ListboxOption,
+  type OptionType,
+} from '@/components/Listbox/ListboxOption'
+import {
+  ListboxOptions,
+  type ListboxOptionsProps,
+} from '@/components/Listbox/ListboxOptions'
+
+export type FormListboxProps = {
+  value: OptionType
+  onChange: (value: OptionType) => void
+  onBlur: () => void
+  options: Array<OptionType>
+  placeholder: string
+  icon?: IconProps['component']
+  buttonWidth?: `w-${keyof typeof theme.spacing}`
+  optionsWidth?: ListboxOptionsProps['width']
+  optionsPosition?: ListboxOptionsProps['position']
+  disabled?: boolean
+} & FormFieldProps
+
+export function FormListbox({
+  label,
+  hideLabel,
+  error,
+  options,
+  value,
+  placeholder,
+  icon,
+  buttonWidth,
+  optionsWidth,
+  optionsPosition,
+  ...rest
+}: FormListboxProps) {
+  return (
+    <FormField label={label} hideLabel={hideLabel} error={error}>
+      <Listbox
+        {...rest}
+        as="div"
+        value={value}
+        invalid={Boolean(error)}
+        className={buttonWidth}
+      >
+        <ListboxButton
+          text={value?.name || placeholder}
+          leadingIcon={icon}
+          hasError={Boolean(error)}
+        />
+        <ListboxOptions width={optionsWidth} position={optionsPosition}>
```

Comment:
why do we pass `optionsWidth`? 🤔 Also do we want the `listboxOptions` to have the same width as `listboxButton` in the project form? 

![Screenshot 2024-11-21 at 17 04 37](https://github.com/user-attachments/assets/2f51e357-ebdb-4e6d-8f06-bd1a30d31be4)

Reply:
Yes, that's a good point. We discussed it with @mirhamasala, and we will make `ListboxOptions` as wide as the content.

So we won't be passing `optionsWidth` anymore :)